src/main/java/com/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -38,4 +38,17 @@ public BookDto findById(Long id) {
         return bookMapper.toDto(book);
 
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        findById(id);
+        Book book = bookMapper.toBook(requestDto);

```

Comment:
You can create another method in BookMapper, to update book from dto.
It these method you just change it to entity and then back to dto, but there is no update